### PR TITLE
Regroup convinient functions for tests into utils package

### PR DESCRIPTION
**Pull request by phacops**
Solve https://github.com/pachyderm/pfs/issues/72.

So far, only 1 function was identified as a duplicate accross tests.

### CODE REVIEW

File: lib/shard/shard_test.go [hunk 1]
```diff
@@ -43,7 +44,7 @@ func TestBasic(t *testing.T) {
 		RunWorkload(s.URL, facts, t)
 		return true
 	}
-	if err := quick.Check(f, &quick.Config{MaxCount: 5}); err != nil {
+	if err := quick.check(f, &quick.Config{MaxCount: 5}); err != nil {
```
Comment (jdoliner) [inline lib/shard/shard_test.go]: This should still be `Check` I think.

**Comment by jdoliner** [comment]
Looks like a good direction.

How do we feel about a `utils` package? We already had one so this pr isn't to blame but does
anyone know where other projects normally put code like this?

Also we might want to look at moving the other functions in `lib/shard/test_helpers.go` somewhere.
It might make sense to move them over to a bindings library which would serve as go bindings to our api.

**Comment by phacops** [comment]
I usually try to not have a package where I put random convenient functions like this because it can get messy pretty quickly (we may put very unrelated functions in there). A better solution might be to specialize the package so it has a specific purpose (compared to a "good-for-everything" package).

I'll take a look to `lib/shard/test_helpers.go`.

File: lib/shard/shard_test.go (hunk 1 above)
Comment (phacops) [inline lib/shard/shard_test.go]: True.

Also, I want to rename `check` (to check a test) with `CheckTest`. Might be easier to not mistake this function for a builtin functions.

**Comment by peter-edge** [comment]
I don't know if we need this at all:

- standard is just to use built-in, no helpers, or if you really need a t.Fatal helper, use testify https://github.com/stretchr/testify
- I don't know the answer to this, but I think that if a file does not end in _test.go, then dependences will be compiled into a main binary, which would mean a main binary would rely on the testing package

**Comment by phacops** [comment]
TL;DR: get rid of test utils or make a test utils package.

`testify` looks cool, but wouldn't it just more test helpers instead of just the one we have here? If we separate test utils and regular utils, I think it'll be identical to have our own `check` function. We shouldn't use any helper then.

Concerning the 2nd bullet point, in this state, you're right, `testing` will be compiled in the binary. If we separate into our own testing package, this package would only be included by files ending by `_test.go` and this wouldn't be a problem.

**Comment by peter-edge** [comment]
Ya so I would just get rid of `check`

In general, I may be in the minority but I actually prefer:

```
if err != nil {
  t.Fatal(...)
}
```

to using wrapper functions. It encourages more fine-grained control over tests (ie `Error` vs `Fatal`), and I use `Errorf/Fatalf` a lot for better messages (you can have `check` and `checkf` too, but again ya).

If we still wanted this, then using `testify` is pretty standard and gets rid of needing a special testing package in pfs.

I don't think this one function deserves making a special package in pfs for

**Comment by peter-edge** [comment]
Hey man,

Super appreciate all this. So right now, we're changing around a bunch of stuff with how we're branching, and the merge will be super painful, plus there's a couple little things here - we feel bad but we're going to close this for now, and let's discuss it more in https://github.com/pachyderm/pfs/issues/72? I feel super bad about closing this but I think you'll like what we're doing and agree it'll be easier for you and for us :)
